src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(health, name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!animal.hidden && animal instanceof Herbivore) {
```

Comment:
You need to check if animal is instanceof Herbivore and only after this check `hidden` flag.

```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
```